### PR TITLE
VAF plot wrapped into react component

### PR DESCRIPTION
**Pull request by adamabeshouse**
# What? Why?
Fix # .

Changes proposed in this pull request:
- a
- b

# Checks
- [ ] Follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Follows the [Airbnb React/JSX Style guide](https://github.com/airbnb/javascript/tree/master/react).
- [ ] Make sure your commit messages end with a Signed-off-by string (this line
  can be automatically added by git if you run the `git-commit` command with
  the `-s` option)

# Any screenshots or GIFs?
If this is a new visual feature please add a before/after screenshot or gif
here with e.g. [GifGrabber](http://www.gifgrabber.com/).

# Notify reviewers
Read our [Pull request merging
policy](../CONTRIBUTING.md#pull-request-merging-policy). If you are part of the
cBioPortal organization, notify the approprate team (remove inappropriate):

@cBioPortal/frontend

If you are not part of the cBioPortal organization look at who worked on the
file before you. Please use `git blame <filename>` to determine that
and notify them here:

### CODE REVIEW

File: src/pages/patientView/PatientViewPage.tsx [hunk 1]
```diff
@@ -57,9 +60,10 @@ export default class PatientViewPage extends React.Component<IPatientViewPagePro
     }
 
     public render() {
-        return (
-            <div>
-                <ClinicalInformationContainer />
+        return (<div>
+            <ClinicalInformationContainer />
+                <ThumbnailExpandVAFPlot data= { vafPlotMockData } colors ={ vafPlotMockColors } labels= { vafPlotMockLabels }
```
Comment (alisman) [inline src/pages/patientView/PatientViewPage.tsx]: minor,  but kinda strange spacing around those attibutes and = sings

File: src/pages/patientView/vafPlot/ThumbnailExpandVAFPlot.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            show_controls: true,
+            nolegend: false,
+            init_show_histogram: true,
+            init_show_curve: true,
+        };
+
+        return (<span>
+            <Tooltip placement={this.props.overlayPlacement} trigger={['hover', 'focus']} overlay={<VAFPlot {...expandedProps}/>}
+                    arrowContent={<div className="rc-tooltip-arrow-inner"></div>}
+                    destroyTooltipOnHide={false}>
```
Comment (alisman) [inline src/pages/patientView/vafPlot/ThumbnailExpandVAFPlot.tsx]: i think destroy:true is generally a good idea unless there's some big cost in reproducing the tooltip

File: src/pages/patientView/vafPlot/ThumbnailExpandVAFPlot.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                <span><VAFPlot key="thumbnail" ref="thumbnail" {...thumbnailProps}/></span>
```
Comment (alisman) [inline src/pages/patientView/vafPlot/ThumbnailExpandVAFPlot.tsx]: what's reason for wrapping span here?

File: src/pages/patientView/vafPlot/ThumbnailExpandVAFPlot.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
+import * as React from 'react';
+import Tooltip from 'rc-tooltip';
+import 'rc-tooltip/assets/bootstrap_white.css';
+import {VAFPlot, IVAFPlotProps} from './VAFPlot';
+
+export type IThumbnailExpandVAFPlotProps = {
```
Comment (alisman) [inline src/pages/patientView/vafPlot/ThumbnailExpandVAFPlot.tsx]: should all of these be optional? seems like data at least is necessary

File: src/pages/patientView/vafPlot/VAFPlot.tsx [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+            } else {
+                $(this.div).find('.viz_curve').hide();
+            }
+        }
+
+        return (nextState.show_histogram !== this.state.show_histogram) || (nextState.show_curve !== this.state.show_curve);
+    }
+
+    toggleShowHistogram(evt) {
+        // setState not necessarily synchronous, so to send the new value to callback we need to do this
```
Comment (alisman) [inline src/pages/patientView/vafPlot/VAFPlot.tsx]: not sure what this comment means

File: src/pages/patientView/vafPlot/VAFPlot.tsx [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+        this.setState({show_curve: new_show_curve});
+    }
+
+    render() {
+        let controls = [<label key="histogram-toggle"><input ref={(input) => { this.histogram_checkbox = input; }}
+                                      type="checkbox" checked={ this.state.show_histogram } onChange= { this.toggleShowHistogram }/>histogram</label>,
+                        <label key="curve-toggle"><input ref={(input) => { this.curve_checkbox = input; }}
+                                      type="checkbox" checked={ this.state.show_curve } onChange = { this.toggleShowCurve } />density estimation</label>];
+        return (
+            <div style={{display: "inline"}} ref={(div) => { this.div = div; }}>
```
Comment (alisman) [inline src/pages/patientView/vafPlot/VAFPlot.tsx]: layout conditions seems kinda funky.   maybe just avoid rendering div when there are no controls.  also, why inline but then <br> tag?

File: src/pages/patientView/vafPlot/VAFPlot.tsx [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+                { ( this.props.show_controls ? controls : [] ) }
+                {( this.props.show_controls ? <br/> : [])}
+            </div>
+        );
+    }
+}
+
+VAFPlot.defaultProps = {
```
Comment (alisman) [inline src/pages/patientView/vafPlot/VAFPlot.tsx]: does this work with typescript?  I think i tried it and was getting ts error.  
I think i found that with ts, default props can be handled with static member of component class:

 public static defaultProps: PageProps = {
        foo: "default"
    };

File: src/pages/patientView/vafPlot/VAFPlot.tsx [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+        this.toggleShowCurve = this.toggleShowCurve.bind(this);
+    }
+
+    componentDidMount() {
+        AlleleFreqPlotMulti(this, this.props.data, this.state.options, this.props.order, this.props.colors, this.props.labels);
+    }
+
+    shouldComponentUpdate(nextProps, nextState) {
+        // Hack around React: handle updating here, and return false
+        if (this.histogram_checkbox) {
```
Comment (alisman) [inline src/pages/patientView/vafPlot/VAFPlot.tsx]: if we determine that rendering AlleleFreqPlotMulti is not prohibitively expensive, we could err on the side of just fully re-rendering it (i.e. in render method).  That way we can avoid these kind of shenanigans and be doing real React.   Basically my point is that we should actually be biased toward real React flow in cases where we're certain there isn't a performance issue and shouldComponentUpdate conditions are simple.  We can discuss.

Comment (adamabeshouse) [inline src/pages/patientView/vafPlot/VAFPlot.tsx]: good idea

File: src/pages/patientView/PatientViewPage.tsx (hunk 1 above)
Comment (adamabeshouse) [inline src/pages/patientView/PatientViewPage.tsx]: good call

File: src/pages/patientView/vafPlot/ThumbnailExpandVAFPlot.tsx (hunk 2 above)
Comment (adamabeshouse) [inline src/pages/patientView/vafPlot/ThumbnailExpandVAFPlot.tsx]: done

File: src/pages/patientView/vafPlot/ThumbnailExpandVAFPlot.tsx (hunk 3 above)
Comment (adamabeshouse) [inline src/pages/patientView/vafPlot/ThumbnailExpandVAFPlot.tsx]: oops, legacy

File: src/pages/patientView/vafPlot/ThumbnailExpandVAFPlot.tsx (hunk 4 above)
Comment (adamabeshouse) [inline src/pages/patientView/vafPlot/ThumbnailExpandVAFPlot.tsx]: good point, data should be necessary

File: src/pages/patientView/vafPlot/VAFPlot.tsx [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+                top: this.props.margin_top,
+                left: this.props.margin_left,
+                right: this.props.margin_right,
+                bottom: this.props.margin_bottom
+            },
+            width: this.props.width,
+            height: this.props.height
+        };
+        // margin bottom must be computed before height because height computation makes use of it
+        options.margin.bottom = (isNaN(options.margin.bottom) ? (30 + (label_dist_to_axis / 2)) : options.margin.bottom);
```
Comment (adamabeshouse) [inline src/pages/patientView/vafPlot/VAFPlot.tsx]: these are used here actually, it's one of the ways we set up the thumbnail and the expanded version to be different.

But I've replaced isNaN with an undefined check.

File: src/pages/patientView/vafPlot/VAFPlot.tsx (hunk 5 above)
Comment (adamabeshouse) [inline src/pages/patientView/vafPlot/VAFPlot.tsx]: oof, more legacy/dev code. sorry

File: src/pages/patientView/vafPlot/VAFPlot.tsx (hunk 6 above)
Comment (adamabeshouse) [inline src/pages/patientView/vafPlot/VAFPlot.tsx]: The imperative rendering method `AlleleFreqPlotMulti` will populate this `div`

File: src/pages/patientView/vafPlot/VAFPlot.tsx (hunk 7 above)
Comment (adamabeshouse) [inline src/pages/patientView/vafPlot/VAFPlot.tsx]: good idea, I'll do this

Comment (adamabeshouse) [inline src/pages/patientView/vafPlot/VAFPlot.tsx]: but if I type the variable (`: PageProps`) then I have to include values for mandatory members, which is kind of the opposite of defaultProps haha.. so I'm leaving it untyped

File: src/pages/patientView/vafPlot/VAFPlot.tsx (hunk 9 above)
Comment (alisman) [inline src/pages/patientView/vafPlot/VAFPlot.tsx]: Sorry, I meant is it likely to be used anywhere else?

File: src/pages/patientView/vafPlot/VAFPlot.tsx (hunk 7 above)
Comment (adufilie) [inline src/pages/patientView/vafPlot/VAFPlot.tsx]: Use Partial<PageProps>

File: src/pages/patientView/vafPlot/VAFPlot.tsx (hunk 9 above)
Comment (adamabeshouse) [inline src/pages/patientView/vafPlot/VAFPlot.tsx]: I'm saying that it's not likely to be used elsewhere on the site, but it's used twice, once for the thumbnail, once for the expanded view, and these options need to be configurable to configure each of those cases.

**Pull request closed without merging**